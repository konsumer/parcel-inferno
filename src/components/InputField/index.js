import { Component } from "inferno";

class InputField extends Component {
    constructor(props) {
        super(props);
        this.state = { value: props.value };
    }

  handleChange = event => {
      this.setState({ value: event.target.value });
  };

  render() {
      return <input type="text" value={this.state.value} onInput={this.handleChange} />;
  }
}

export default InputField;
