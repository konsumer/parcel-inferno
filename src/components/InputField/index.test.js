import InputField from ".";
import { renderToSnapshot } from "inferno-test-utils";

it("renders without crashing", () => {
  expect(renderToSnapshot(<InputField value={"test value"} />)).toMatchSnapshot();
});
