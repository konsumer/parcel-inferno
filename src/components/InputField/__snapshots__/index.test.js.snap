// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<input
  onInput={[Function]}
  type="text"
  value="test value"
/>
`;
